File: Selenium/Project/WebsiteHeading.java
package Project;


import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class WebsiteHeading {
        private WebDriver driver;
        private String baseUrl = "https://alchemy.hguy.co/jobs/";

        @BeforeClass
        public void setUp() {
            System.setProperty("webdriver.gecko.driver",  "C://Users//001ZXG744//IdeaProjects//geckodriver.exe");
            driver = new FirefoxDriver();
            driver.get(baseUrl);
        }

        @Test
        public void testWebsiteHeading() {
            // Get the heading of the webpage
            WebElement headingElement = driver.findElement(By.xpath("//h1[@class='entry-title']"));
            String actualHeading = headingElement.getText();

            // Verify the heading matches the expected heading
            String expectedHeading = "Welcome to Alchemy Jobs";
            Assert.assertEquals(actualHeading, expectedHeading, "Website heading does not match the expected heading.");
        }

        @AfterClass
        public void tearDown() {
            driver.quit();
        }
    }

File: Selenium/Project/WebsiteHeading.py
import unittest
from selenium import webdriver
from selenium.webdriver.common.by import By

class WebsiteHeading(unittest.TestCase):
    def setUp(self):
        self.driver = webdriver.Firefox(executable_path="C:/Users/001ZXG744/IdeaProjects/geckodriver.exe")
        self.base_url = "https://alchemy.hguy.co/jobs/"

    def test_website_heading(self):
        # Navigate to the base URL
        self.driver.get(self.base_url)

        # Get the heading of the webpage
        heading_element = self.driver.find_element(By.XPATH, "//h1[@class='entry-title']")
        actual_heading = heading_element.text

        # Verify the heading matches the expected heading
        expected_heading = "Welcome to Alchemy Jobs"
        self.assertEqual(actual_heading, expected_heading, "Website heading does not match the expected heading.")

    def tearDown(self):
        self.driver.quit()

if __name__ == "__main__":
    unittest.main()
